confluent_kafka/schema_registry/protobuf.py

```diff
@@ -0,0 +1,437 @@
+
+    Returns:
+        str: Base64 encoded FileDescriptor
+
+    """
+    return base64.standard_b64encode(proto_file.serialized_pb).decode('ascii')
+
+
+class ProtobufSerializer(object):
+    """
+    ProtobufSerializer serializes objects in the Confluent Schema Registry
+    binary format for Protobuf.
+
+    ProtobufSerializer configuration properties:
+    +-----------------------+----------+------------------------------------------------------+
+    | Property Name         | Type     | Description                                          |
+    +=======================+==========+======================================================+
+    |                       |          | Registers schemas automatically if not               |
+    | auto.register.schemas | bool     | previously associated with a particular subject.     |
+    |                       |          | Defaults to True.                                    |
+    +-----------------------|----------+------------------------------------------------------+
+    |                       |          | Callable(SerializationContext, str) -> str           |
+    |                       |          |                                                      |
+    | subject.name.strategy | callable | Instructs the ProtobufSerializer on how to construct |
+    |                       |          | Schema Registry subject names.                       |
+    |                       |          | Defaults to topic_subject_name_strategy.             |
+    +-----------------------+----------+------------------------------------------------------+
+
+    Schemas are registered to namespaces known as Subjects which define how a
+    schema may evolve over time. By default the subject name is formed by
+    concatenating the topic name with the message field separated by a hyphen.
+
+    i.e. {topic name}-{message field}
+
+    Alternative naming strategies may be configured with the property
+    `subject.name.strategy`.
+
+    Supported subject name strategies:
+    +--------------------------------------+------------------------------+
+    | Subject Name Strategy                | Output Format                |
+    +======================================+==============================+
+    | topic_subject_name_strategy(default) | {topic name}-{message field} |
+    +--------------------------------------+------------------------------+
+    | topic_record_subject_name_strategy   | {topic name}-{record name}   |
+    +--------------------------------------+------------------------------+
+    | record_subject_name_strategy         | {record name}                |
+    +--------------------------------------+------------------------------+
+
+    See ``Subject name strategy`` for additional details.
+
+    Args:
+        schema_registry_client (SchemaRegistryClient): Schema Registry
+            client instance.
+
+        descriptor (Descriptor): Protobuf Message Descriptor.
```

Comment:
Why do you need to pass the descriptor to the serializer constructor? The serializer only accepts classes generated by `protoc`, so you should be able to get it on the fly.

Reply:
Originally @magnus requested that serializers be 1:1 with the type they are serializing. This I believe was based on how Avro works though since the Schema is not bound to the instance as is the case with Protobuf. With that said this serializer knows how to  serialize one type of proto message and one only its in the constructor. If we want to be more flexible, which is substantially easier with protobuf than avro (since we use generic types) that's more than okay with me.

Reply:
consistency is an almost insignificant priority here imo.

Reply:
FWIW I agree, but I'll wait for us to all come to agreement before jumping because I don't want to do it twice.

Reply:
Done, will follow up with Many:One Protobuf serializer

Reply:
Resolving based  on the decision to allow  for multiple types to be handled by the same serializer. We can discuss the implementation in the follow up PR.